src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -342,10 +343,22 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
 			if (metadata != null) {
 				HashMap<String, String> metadataMap = blob.getMetadata();
 				for (AzureBlobMetadataPair pair : metadata) {
-					metadataMap.put(
-						Util.replaceMacro(pair.getKey(), env),
-						Util.replaceMacro(pair.getValue(), env)
-					);
+					final String resolvedKey = Util.replaceMacro(pair.getKey(), env);
+					final String resolvedValue = Util.replaceMacro(pair.getValue(), env);
+
+					// Azure does not allow null, empty or whitespace metadata key
+					if (resolvedKey == null || resolvedKey.trim().length() == 0) {
```

Comment:
Seems there is already a `Utils.isNullOrEmpty()` method to check whether a string is null or empty. There're also some places that use `StringUtils.isBlank()`. Can we consolidate them to use the same method?